src/server.cpp

```diff
@@ -20,19 +23,55 @@ void Server::start_receive()
         std::bind(&Server::handle_receive, this, pls::_1, pls::_2));
 }
 
-void Server::handle_receive(const error_code& error, std::size_t /*bytes_transferred*/)
+void Server::handle_receive(const error_code& error, std::size_t bytes)
 {
-    if (!error || error == boost::asio::error::message_size)
+    if (!error)
```

Comment:
I see that in case of an error you do nothing, just exit the function. It would be nice to write something about the error, do some logging.